test/test_http_stress.py

```diff
@@ -0,0 +1,153 @@
+        return
+
+    def finish(self):
+        if self.__loggerHandle is not None:
+            self.__loggerHandle.close()
+        super().finish()
+
+
+def set_up_local_server_in_thread():
+    try:
+        Handler = partial(FileLoggerSimpleHTTPRequestHandler, logfile=None)
+        socketserver.TCPServer.allow_reuse_address = True
+
+        server = socketserver.TCPServer(("", 0), Handler)
+        server_addr = f"{server.server_address[0]}:{server.server_address[1]}"
+        server_thread = threading.Thread(target=server.serve_forever)
+        server_thread.start()
+
+        # Wait a bit for the server to come up
+        time.sleep(3)
+
+        return server_thread, server_addr, server
+    except Exception:
+        raise
+
+
+def create_temp_files_for_serving(tmp_dir, file_count, file_size, file_url_template, i):
+    furl_local_file = os.path.join(tmp_dir, "urls_list")
+    f = os.path.join(tmp_dir, f"webfile_test_{i}.data")  # One generated file will be read repeatedly
+    write_chunk = 1024 * 1024 * 16
+    rmn_size = file_size
+    with open(f, "ab+") as fout:
+        while rmn_size > 0:
+            fout.write(os.urandom(min(rmn_size, write_chunk)))
+            rmn_size = rmn_size - min(rmn_size, write_chunk)
+    with open(furl_local_file, "w") as fsum:
+        for _ in range(file_count):
+            fsum.write(file_url_template.format(num=i))
+            print(f"URL written: {file_url_template.format(num=i)}")
+
+
+class TestHttpStress(expecttest.TestCase):
+    __server_thread: threading.Thread
+    __server_addr: str
+    __server: socketserver.TCPServer
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            (cls.__server_thread, cls.__server_addr, cls.__server) = set_up_local_server_in_thread()
+            print(f"\nServer address: {cls.__server_addr}")
+        except Exception as e:
+            warnings.warn(
+                "TestHttpStress could\
+                          not set up due to {}".format(
+                    str(e)
+                )
+            )
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cls.__server.shutdown()
+            cls.__server_thread.join(timeout=15)
+        except Exception as e:
+            warnings.warn(
+                "TestHttpStress could\
+                           not tear down (clean up temp directory or terminate\
+                           local server) due to {}".format(
+                    str(e)
+                )
+            )
+
+    def _http_test_base(self, test_file_size, test_file_count, timeout=None, chunk=None, i=0):
+        def _get_data_from_tuple_fn(data):
+            return data[1]
+
+        with tempfile.TemporaryDirectory(dir=os.getcwd()) as tmpdir:
+            # create tmp dir and files for test
+            base_tmp_dir = os.path.basename(os.path.normpath(tmpdir))
+            url = "http://{server_addr}/{tmp_dir}/webfile_test_{num}.data\n"
```

Comment:
Have you tried to use `os.join` here to validate if it's the path causes the windows issue?